Lesson19/gozde/recursion.js

```diff
@@ -0,0 +1,210 @@
+// See completed in the lesson examples in the lesson files folder: https://github.com/frontend-from-0/group091123-files
+
+// 4. Write a recursive function to remove all occurrences of a specified character from a string.
+const sentenceEx5 = 'Hello, how are you?';
+const charToRemove = 'o';
+
+function removeCharRecursive(sentence, charToRemove) {
+    if (sentence.length === 0) {
+        return '';
+    }
+    
+    if (sentence[0] === charToRemove) {
+        return removeCharRecursive(sentence.slice(1), charToRemove);
+    } else {
+        return sentence[0] + removeCharRecursive(sentence.slice(1), charToRemove);
+    }
+}
+
+const sentenceEx5 = 'Hello, how are you?';
+const charToRemove = 'o';
+const result = removeCharRecursive(sentenceEx5, charToRemove);
+
+console.log(result);
+
+// 5. Write a recursive function to check if an array includes a specific value.
+const numbers = [1, 2, 3, 4, 5];
+const valueToCheck = 3;
+
+function includesValueRecursive(array, valueToCheck) {
+    if (array.length === 0) {
+        return false;
+    }
+    
+    if (array[0] === valueToCheck) {
+        return true;
+    } else {
+        return includesValueRecursive(array.slice(1), valueToCheck);
+    }
+}
+
+const numbers = [1, 2, 3, 4, 5];
+const valueToCheck = 3;
+const result = includesValueRecursive(numbers, valueToCheck);
+
+console.log(result);
+
+// 6. Write a recursive function to flatten an object with nested objects into a single-level object.
+
+function flattenObject(obj, parentKey = '') {
+    let result = {};
+    for (let key in obj) {
+        if (typeof obj[key] === 'object' && !Array.isArray(obj[key])) {
+            const flattened = flattenObject(obj[key], `${parentKey}${key}.`);
+            result = { ...result, ...flattened };
+        } else if (Array.isArray(obj[key])) {
+            obj[key].forEach((item, index) => {
+                result[`${parentKey}${key}[${index}]`] = item;
+            });
+        } else {
+            result[`${parentKey}${key}`] = obj[key];
+        }
+    }
+    return result;
+}
+
+const nestedObj = {
+    name: 'John',
+    age: 30,
+    address: {
+        street: '123 Main St',
+        city: 'New York',
+        country: 'USA',
+    },
+    hobbies: ['reading', 'coding'],
+    social: {
+        twitter: '@johndoe',
+        linkedIn: 'johndoe',
+    },
+};
+
+const flattenedObj = flattenObject(nestedObj);
+console.log(flattenedObj);
+
+      
+
+// 7. Write a recursive function to find the maximum depth of a nested object.
+
+function maxDepth(obj) {
```

Comment:
The issue in this function is that the function and and a variable name inside of it are the same (maxDepth). When calling function recursively, compiler does not recognize 'maxDepth' as function but it thinks it's a variable (`const depth = 1 + maxDepth(obj[key]);`). 

Solution: rename the function to e.g. 'calculateMaxDepth', make sure that you rename all the occurances.